File: develop/learning-paths/mobile-sdk-android/code/02-retrieving-and-displaying-portlet-data/LiferayGuestbook/app/src/main/java/com/liferay/docs/liferayguestbook/model/EntryModel.java
package com.liferay.docs.liferayguestbook.model;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.Date;

public class EntryModel implements Serializable {

    private long _entryId;
    private long _groupId;
    private long _companyId;
    private long _userId;
    private String _userName;
    private long _createDate;
    private long _modifiedDate;
    private String _name;
    private String _email;
    private String _message;
    private long _guestbookId;

    public EntryModel(JSONObject json) throws JSONException {

        _entryId = json.getLong("entryId");
        _groupId = json.getLong("groupId");
        _companyId = json.getLong("companyId");
        _userId = json.getLong("userId");
        _userName = json.getString("userName");
        _createDate = json.getLong("createDate");
        _modifiedDate = json.getLong("modifiedDate");
        _name = json.getString("name");
        _email = json.getString("email");
        _message = json.getString("message");
        _guestbookId = json.getLong("guestbookId");
    }

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof GuestbookModel)) {
            return false;
        }

        EntryModel entry = (EntryModel)obj;

        return (_entryId == entry.getEntryId());
    }

    public long getEntryId() {
        return _entryId;
    }

    public long getGroupId() {
        return _groupId;
    }

    public long getCompanyId() {
        return _companyId;
    }

    public long getUserId() {
        return _userId;
    }

    public String getUserName() {
        return _userName;
    }

    public Date getCreateDate() {
        Date createDate = new Date(_createDate);
        return createDate;
    }

    public Date getModifiedDate() {
        Date modifiedDate = new Date(_modifiedDate);
        return modifiedDate;
    }

    public String getName() {
        return _name;
    }

    public String getEmail() {
        return _email;
    }

    public String getMessage() {
        return _message;
    }

    public long getGuestbookId() {
        return _guestbookId;
    }
}

File: develop/tutorials/code/mobile/ios/screenlets-app/Liferay Guestbook/model/EntryModel.swift
import UIKit

@objc public class EntryModel: NSObject {
    
    public let attributes: [String:AnyObject]
    
    public var entryId: Int64 {
        return attributes["entryId"]?.int64Value ?? 0
    }
    
    public var groupId: Int64 {
        return attributes["groupId"]?.int64Value ?? 0
    }
    
    public var companyId: Int64 {
        return attributes["companyId"]?.int64Value ?? 0
    }
    
    public var userId: Int64 {
        return attributes["userId"]?.int64Value ?? 0
    }
    
    public var userName: String {
        return attributes["userName"] as? String ?? ""
    }
    
    public var createDate: Int64 {
        return attributes["createDate"]?.int64Value ?? 0
    }
    
    public var modifiedDate: Int64 {
        return attributes["modifiedDate"]?.int64Value ?? 0
    }
    
    public var name: String {
        return attributes["name"] as? String ?? ""
    }
    
    public var email: String {
        return attributes["email"] as? String ?? ""
    }
    
    public var message: String {
        return attributes["message"] as? String ?? ""
    }
    
    public var guestbookId: Int64 {
        return attributes["guestbookId"]?.int64Value ?? 0
    }
    
    //MARK: Initializer
    
    public init(attributes: [String:AnyObject]) {
        self.attributes = attributes
    }
    
}
